### ISSUE TITLE
[bug] `@eslint-react/naming-convention/component-name` does not report error if there is a valid component before an invalid one

### ISSUE THREAD

**Issue by ariallas** [Rel1cx/eslint-react#1176]
### Describe the bug

`@eslint-react/naming-convention/component-name` rule does not report errors if there is a component with a valid name declared before the one with invalid name.

It seems like there should be a `continue` here instead of `return`:
https://github.com/Rel1cx/eslint-react/blob/8041f451e11034fdc2a21acccbd823e7715d21c6/packages/plugins/eslint-plugin-react-naming-convention/src/rules/component-name.ts#L118

### Reproduction

```ts
// eslint.config.js
  rules: {
    "@eslint-react/naming-convention/component-name": ["warn", "PascalCase"],
  },
```

Example with rule working correctly:
```ts
import React from "react";

// Correctly reports "A component name 'My_component' does not match PascalCase"
function My_component() {
  return null;
}

function AnotherComponent() {
  return null;
}
```

Example with rule not working correctly:
```ts
import React from "react";

function AnotherComponent() {
  return null;
}

// Does not report component-name warning
function My_component() {
  return null;
}
```

### Expected behavior

`My_component` is reported for "component-name" rule, regardless of component order in the file.

### Platform and versions

```Text
node v24.1.0
"@eslint-react/eslint-plugin": "^1.52.5",
```

### Stack trace

```Text

```
